### ISSUE TITLE
Add site-dependent QNs numbers

### ISSUE THREAD

**Issue by LHerviou** [ITensor/ITensorInfiniteMPS.jl#47]
The final step I need for my FQHE code is to implement site-dependent QNs.
I managed to make it work with a little bit of dirty code on ITensors  for finite DMRG, so it should be also workable.

Now, it would be logical to implement this at the level of the CelledVectors. 
One possibility is to add a functional component to the structure (e.g. named QN_translation_operation) which would be called and applied on the QNs. To make the behavior transparent, I could make its default value "nothing", and add a few if lines in the code that only applies the function if it is assigned/not nothing.

A description of the implementation of the symmetry I need is given in the Supplemental Materials of https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.110.236801. 
Basically, if I note the shifted charge conservation  C_n = q N_n  - p (for site n at filling p/q), the local momentum operator is K_n = n C_n  + s, where s is a constant shift that is fixed at the creation of the MPS. For a well chosen s, the transformation rules for all charge legs would be (c, k) -> (c, k + nsites(\psi) c) for all legs.

**Comment by LHerviou** [comment]
That seems to fit exactly what I had in mind.
I will start having a look at it.

**Comment by mtfishman** [comment]
Looks like that error shows that you are using the site type `"Fermions"` instead of `"Fermion"`. This works:
```julia
julia> using ITensors

julia> s = siteind("Fermion")
(dim=2|id=199|"Fermion,Site")

julia> @show op("N", s);
op("N", s) = ITensor ord=2
Dim 1: (dim=2|id=199|"Fermion,Site")'
Dim 2: (dim=2|id=199|"Fermion,Site")
NDTensors.Dense{Float64, Vector{Float64}}
 2×2
 0.0  0.0
 0.0  1.0
```

**Comment by LHerviou** [comment]
I was blind. Thanks for the fix, the generalized cellvectors are working.

I will make a PR once the current one is synchronized, probably also including a generalized subspace expansion I needed for my computations.

**Comment by mtfishman** [comment]
Amazing, thanks!

**Comment by LHerviou** [comment]
A quick comment: some default functions for translatecells are not implemented apparently (a colleague was struggling) => probably a consequence of the name change. 
I will try to make a PR for that tomorrow at the latest.
